llama-index-integrations/tools/llama-index-tools-agentql/llama_index/tools/agentql/agentql_rest_api_tool/base.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional
+import os
+
+from llama_index.core.tools.tool_spec.base import BaseToolSpec
+
+from llama_index.tools.agentql.const import (
+    DEFAULT_API_TIMEOUT_SECONDS,
+    DEFAULT_IS_STEALTH_MODE_ENABLED,
+    DEFAULT_WAIT_FOR_PAGE_LOAD_SECONDS,
+    DEFAULT_IS_SCROLL_TO_BOTTOM_ENABLED,
+    DEFAULT_RESPONSE_MODE,
+    DEFAULT_IS_SCREENSHOT_ENABLED,
+)
+from llama_index.tools.agentql.messages import UNSET_API_KEY_ERROR_MESSAGE
+from llama_index.tools.agentql.utils import _aload_data
+
+
+class AgentQLRestAPIToolSpec(BaseToolSpec):
+    """
+    AgentQL Rest API Tool Spec.
+    """
+
+    spec_functions = [
+        "extract_web_data_with_rest_api",
+    ]
+
+    def __init__(
+        self,
+        timeout: Optional[int] = DEFAULT_API_TIMEOUT_SECONDS,
```

Comment:
all these field don't have Optional type, they are optional by nature, because they have default values. But it's not allowed to pass None as their values